Core/src/core/smallEvents/lottery.ts

```diff
@@ -0,0 +1,193 @@
+import {TravelTime} from "../maps/TravelTime";
+import {Effect} from "../../../../Lib/src/enums/Effect";
+import {NumberChangeReason} from "../../../../Lib/src/constants/LogsConstants";
+import {RandomUtils} from "../utils/RandomUtils";
+
+type LotteryProperties = {
+	successRate: {
+		[lotteryLevel in LotteryLevelKey]: number
+	},
+	coefficients: {
+		[lotteryLevel in LotteryLevelKey]: number
+	},
+	lostTime: number,
+}
+
+type LotteryLevelKey = "hard" | "medium" | "easy";
+
+async function effectIfGoodRisk(levelKey: LotteryLevelKey, player: Player, dataLottery: LotteryProperties): Promise<number> {
+	if (levelKey !== "easy") {
+		await TravelTime.applyEffect(
+			player,
+			Effect.OCCUPIED,
+			dataLottery.lostTime,
+			new Date(),
+			NumberChangeReason.SMALL_EVENT
+		);
+
+		return dataLottery.lostTime;
+	}
+
+	return 0;
+}
+
+// eslint-disable-next-line max-params
+async function giveRewardToPlayer(
+	rewardType: string,
+	player: Player,
+	coefficient: number,
+	response: DraftBotPacket[],
+	lostTime: number,
+	levelKey: "easy" | "hard" | "medium",
+	guild: Guild
+): Promise<void> {
+	switch (rewardType) {
+	case SmallEventConstants.LOTTERY.REWARD_TYPES.XP:
+		await player.addExperience({
+			amount: SmallEventConstants.LOTTERY.REWARDS.EXPERIENCE * coefficient,
+			response,
+			reason: NumberChangeReason.SMALL_EVENT
+		});
+		response.push(makePacket(SmallEventLotteryWinPacket, {
+			xp: SmallEventConstants.LOTTERY.REWARDS.EXPERIENCE * coefficient,
+			lostTime,
+			level: levelKey
+		}));
+		break;
+	case SmallEventConstants.LOTTERY.REWARD_TYPES.MONEY:
+		await player.addMoney({
+			amount: SmallEventConstants.LOTTERY.REWARDS.MONEY * coefficient,
+			response,
+			reason: NumberChangeReason.SMALL_EVENT
+		});
+		response.push(makePacket(SmallEventLotteryWinPacket, {
+			money: SmallEventConstants.LOTTERY.REWARDS.MONEY * coefficient,
+			lostTime,
+			level: levelKey
+		}));
+		break;
+	case SmallEventConstants.LOTTERY.REWARD_TYPES.GUILD_XP:
+		await guild.addExperience(SmallEventConstants.LOTTERY.REWARDS.GUILD_EXPERIENCE * coefficient, response, NumberChangeReason.SMALL_EVENT);
+		await guild.save();
+		response.push(makePacket(SmallEventLotteryWinPacket, {
+			guildXp: SmallEventConstants.LOTTERY.REWARDS.GUILD_EXPERIENCE * coefficient,
+			lostTime,
+			level: levelKey
+		}));
+		break;
+	case SmallEventConstants.LOTTERY.REWARD_TYPES.POINTS:
+		await player.addScore({
+			amount: SmallEventConstants.LOTTERY.REWARDS.POINTS * coefficient,
+			response,
+			reason: NumberChangeReason.SMALL_EVENT
+		});
+		response.push(makePacket(SmallEventLotteryWinPacket, {
+			points: SmallEventConstants.LOTTERY.REWARDS.POINTS * coefficient,
+			lostTime,
+			level: levelKey
+		}));
+		break;
+	default:
+		throw new Error("lottery reward type not found");
+	}
+}
+
+export const smallEventFuncs: SmallEventFuncs = {
+	canBeExecuted: (player: Player): Promise<boolean> => Promise.resolve(Maps.isOnContinent(player)),
```

Comment:
```suggestion
	canBeExecuted: Maps.isOnContinent,
```
cf https://github.com/DraftBot-A-Discord-Adventure/DraftBot/blob/draftbot-v5/Core/src/core/smallEvents/staffMember.ts